File: 1.two-sum.js
/*
 * @lc app=leetcode id=1 lang=javascript
 *
 * [1] Two Sum
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
function twoSum(nums, target) {
  const map = new Map();

  for (let i = 0; i < nums.length; i++) {
    const cur = nums[i];
    const complementIndex = map.get(target - cur);
    if (complementIndex !== undefined) {
      return [complementIndex, i];
    }

    map.set(cur, i);
  }
}
// @lc code=end

File: 1.two-sum.rs
/*
 * @lc app=leetcode id=1 lang=rust
 *
 * [1] Two Sum
 */

// @lc code=start
impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut map = std::collections::HashMap::new();

        for (i, num) in nums.iter().enumerate() {
            let complement = target - num;
            match map.get(&complement) {
                Some(&index) => {
                    return vec![index as i32, i as i32];
                },
                None => map.insert(num, i),
            };
        }
        vec![]
    }
}
// @lc code=end

